File: backend/config/locations.py
LOCATIONS = {
    "server_room": {
        "label": "Server Room",
        "floor": "main_facility",
        "floor_label": "Main Facility",
        "type": "room",
    },
    "meeting_room": {
        "label": "Meeting Room",
        "floor": "main_facility",
        "floor_label": "Main Facility",
        "type": "room",
    },
    "multi_purpose_room": {
        "label": "Multi-purpose Room",
        "floor": "main_facility",
        "floor_label": "Main Facility",
        "type": "room",
    },
    "gathering_area": {
        "label": "Gathering Area",
        "floor": "main_facility",
        "floor_label": "Main Facility",
        "type": "open_area",
    },
    "conference_room": {
        "label": "Conference Room",
        "floor": "main_facility",
        "floor_label": "Main Facility",
        "type": "room",
    },
    "canteen": {
        "label": "Canteen",
        "floor": "main_facility",
        "floor_label": "Main Facility",
        "type": "open_area",
    },
    "lobby": {
        "label": "Lobby",
        "floor": "main_facility",
        "floor_label": "Main Facility",
        "type": "open_area",
    },

    "ceo_office": {
        "label": "CEO Office",
        "floor": "office_floor",
        "floor_label": "Office Floor",
        "type": "room",
    },
    "manager_office": {
        "label": "Manager Office",
        "floor": "office_floor",
        "floor_label": "Office Floor",
        "type": "room",
    },
    "executive_office": {
        "label": "Executive Office",
        "floor": "office_floor",
        "floor_label": "Office Floor",
        "type": "room",
    },
    "office_area": {
        "label": "Office Area",
        "floor": "office_floor",
        "floor_label": "Office Floor",
        "type": "open_area",
    },
    "command_center": {
        "label": "Command Center",
        "floor": "office_floor",
        "floor_label": "Office Floor",
        "type": "security_area",
    },

    "parking_area": {
        "label": "Parking Area",
        "floor": "basement",
        "floor_label": "Basement / Parking",
        "type": "parking",
    },
    "store_room": {
        "label": "Store Room",
        "floor": "basement",
        "floor_label": "Basement / Parking",
        "type": "room",
    },
}

def is_valid_location(location_key: str) -> bool:
    return location_key in LOCATIONS


def get_location(location_key: str) -> dict | None:
    return LOCATIONS.get(location_key)

File: frontend/src/config/locations.js
export const LOCATION_OPTIONS = [
  { value: "server_room", label: "Server Room" },
  { value: "meeting_room", label: "Meeting Room" },
  { value: "multi_purpose_room", label: "Multi-purpose Room" },
  { value: "gathering_area", label: "Gathering Area" },
  { value: "conference_room", label: "Conference Room" },
  { value: "canteen", label: "Canteen" },
  { value: "lobby", label: "Lobby" },
  { value: "ceo_office", label: "CEO Office" },
  { value: "manager_office", label: "Manager Office" },
  { value: "executive_office", label: "Executive Office" },
  { value: "office_area", label: "Office Area" },
  { value: "command_center", label: "Command Center" },
  { value: "parking_area", label: "Parking Area" },
  { value: "store_room", label: "Store Room" },
];

export const DEFAULT_LOCATION = "server_room";

export function getLocationLabel(locationKey) {
  return (
    LOCATION_OPTIONS.find((location) => location.value === locationKey)?.label ||
    locationKey
  );
}

export function isKnownLocation(locationKey) {
  return LOCATION_OPTIONS.some((location) => location.value === locationKey);
}
